tests/contrib/operators/test_gcp_spanner_operator.py

```diff
@@ -164,6 +170,78 @@ def test_instance_delete_ex_if_param_missing(self, project_id, instance_id, exp_
+        op = CloudSpannerInstanceDatabaseQueryOperator(
+            project_id=PROJECT_ID,
+            instance_id=INSTANCE_ID,
+            database_id=DB_ID,
+            query=INSERT_QUERY,
+            task_id="id"
+        )
+        result = op.execute(None)
+        self.assertIsNone(result)
+
+    @parameterized.expand([
+        ("", INSTANCE_ID, DB_ID, INSERT_QUERY, "project_id"),
+        (PROJECT_ID, "", DB_ID, INSERT_QUERY, "instance_id"),
+        (PROJECT_ID, INSTANCE_ID, "", INSERT_QUERY, "database_id"),
+        (PROJECT_ID, INSTANCE_ID, DB_ID, "", "query"),
+    ])
+    @mock.patch("airflow.contrib.operators.gcp_spanner_operator.CloudSpannerHook")
+    def test_instance_query_ex_if_param_missing(self, project_id, instance_id,
+                                                database_id, query, exp_msg, mock_hook):
+        with self.assertRaises(AirflowException) as cm:
+            CloudSpannerInstanceDatabaseQueryOperator(
+                project_id=project_id,
+                instance_id=instance_id,
+                database_id=database_id,
+                query=query,
+                task_id="id"
+            )
+        err = cm.exception
+        self.assertIn("The required parameter '{}' is empty".format(exp_msg), str(err))
+        mock_hook.assert_not_called()
+
+    @mock.patch("airflow.contrib.operators.gcp_spanner_operator.CloudSpannerHook")
+    def test_instance_query_dml(self, mock_hook):
+        mock_hook.return_value.execute_sql.return_value = None
+        mock_hook.return_value.update_ddl.return_value = None
+        op = CloudSpannerInstanceDatabaseQueryOperator(
+            project_id=PROJECT_ID,
+            instance_id=INSTANCE_ID,
+            database_id=DB_ID,
+            query=INSERT_QUERY,
+            task_id="id"
+        )
+        op.execute(None)
+        mock_hook.assert_called_once_with(gcp_conn_id="google_cloud_default")
+        mock_hook.return_value.execute_sql.assert_called_once_with(
+            PROJECT_ID, INSTANCE_ID, DB_ID, INSERT_QUERY
+        )
+        mock_hook.return_value.update_ddl.assert_not_called()
+
+    @mock.patch("airflow.contrib.operators.gcp_spanner_operator.CloudSpannerHook")
+    def test_instance_query_dml_list(self, mock_hook):
+        mock_hook.return_value.execute_sql.return_value = None
+        mock_hook.return_value.update_ddl.return_value = None
+        op = CloudSpannerInstanceDatabaseQueryOperator(
+            project_id=PROJECT_ID,
+            instance_id=INSTANCE_ID,
+            database_id=DB_ID,
+            query=[INSERT_QUERY, INSERT_QUERY_2],
+            task_id="id"
+        )
+        op.execute(None)
+        mock_hook.assert_called_once_with(gcp_conn_id="google_cloud_default")
+        calls = [
+            mock.call(PROJECT_ID, INSTANCE_ID, DB_ID, INSERT_QUERY),
+            mock.call(PROJECT_ID, INSTANCE_ID, DB_ID, INSERT_QUERY_2),
+        ]
+        mock_hook.return_value.execute_sql.assert_has_calls(calls, any_order=True)
```

Comment:
Can we provide a guarantee that query 2 follows query 1?

Reply:
I've changed this test altogether as it was actually testing an older version of the method.
Sorry about that.